sdks/python/apache_beam/io/gcp/bigtableio.py

```diff
@@ -141,3 +148,127 @@ def expand(self, pvalue):
             | beam.ParDo(_BigTableWriteFn(beam_options['project_id'],
                                           beam_options['instance_id'],
                                           beam_options['table_id'])))
+
+
+class BigtableSource(iobase.BoundedSource):
+  def __init__(self, project_id, instance_id, table_id, filter_=None):
+    """ Constructor of the Read connector of Bigtable
+
+    Args:
+      project_id: [string] GCP Project of to write the Rows
+      instance_id: [string] GCP Instance to write the Rows
+      table_id: [string] GCP Table to write the `DirectRows`
+      filter_: [RowFilter] Filter to apply to columns in a row.
+    """
+    super(self.__class__, self).__init__()
+    self._init({'project_id': project_id,
+                'instance_id': instance_id,
+                'table_id': table_id,
+                'filter_': filter_})
+
+  def __getstate__(self):
+    return self.beam_options
+
+  def __setstate__(self, options):
+    self._init(options)
+
+  def _init(self, options):
+    self.beam_options = options
+    self.table = None
+    self.sample_row_keys = None
+    self.row_count = Metrics.counter(self.__class__.__name__, 'Row count')
+
+
+  def _get_table(self):
+    if self.table is None:
+      self.table = Client(project=self.beam_options['project_id'])\
+                      .instance(self.beam_options['instance_id'])\
+                      .table(self.beam_options['table_id'])
+    return self.table
+
+  def get_sample_row_keys(self):
+    """ Get a sample of row keys in the table.
+
+    The returned row keys will delimit contiguous sections of the table of
+    approximately equal size, which can be used to break up the data for
+    distributed tasks like mapreduces.
+    :returns: A cancel-able iterator. Can be consumed by calling ``next()``
+    			  or by casting to a :class:`list` and can be cancelled by
+    			  calling ``cancel()``.
+    """
+    if self.sample_row_keys is None:
+      self.sample_row_keys = list(self._get_table().sample_row_keys())
+      if self.sample_row_keys[0].row_key != b'':
+        SampleRowKey = namedtuple("SampleRowKey", "row_key offset_bytes")
+        first_key = SampleRowKey(b'', 0)
+        self.sample_row_keys.insert(0, first_key)
+    return self.sample_row_keys
+
+  def get_range_tracker(self, start_position=b'', stop_position=b''):
+    if stop_position == b'':
+      return LexicographicKeyRangeTracker(start_position)
+    else:
+      return LexicographicKeyRangeTracker(start_position, stop_position)
+
+  def estimate_size(self):
+    return self.get_sample_row_keys()[-1].offset_bytes
```

Comment:
Had to add the list() wrapping:
`  def estimate_size(self):
    return list(self.get_sample_row_keys()[-1]).offset_bytes`